tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/Linker.scala

```diff
@@ -224,19 +224,16 @@ final class Linker(semantics: Semantics, outputMode: OutputMode,
           fields += field
 
       // Normal methods
-      case m: MethodDef if m.name.isInstanceOf[Ident] =>
-        if (analyzerInfo.isAnySubclassInstantiated &&
-            analyzerInfo.methodInfos(m.name.name).isReachable) {
-          if (m.body == EmptyTree)
+      case m: MethodDef =>
+        if (analyzerInfo.methodInfos(m.name.name).isReachable) {
```

Comment:
Yes. `isReachable` means that this particular implementation of the method is reachable. It must be kept.